Objects/typeobject.c

```diff
@@ -6665,47 +6671,48 @@ object_set_class(PyObject *self, PyObject *value, void *closure)
         if (oldto->tp_flags & Py_TPFLAGS_INLINE_VALUES) {
             PyDictObject *dict = _PyObject_MaterializeManagedDict(self);
```

Comment:
We should expose and call `materialize_managed_dict_lock_held` instead of `_PyObject_MaterializeManagedDict`.

Stop the world has some sharp edges. It's not safe to acquire locks that might be held by other threads (including through critical sections) in a stop the world context.